packages/kbn-lens-embeddable-utils/config_builder/charts/xy.ts

```diff
@@ -159,16 +170,14 @@ function buildFormulaLayer(
     const resultLayer = {
       ...getFormulaColumn(
         `${ACCESSOR}${i}`,
-        {
-          value: layer.value,
-        },
+        isFormulaValue(layer.value) ? layer.value : { formula: layer.value },
         dataView,
         formulaAPI
       ),
     };
 
     if (layer.xAxis) {
-      const columnName = `${ACCESSOR}${i}_x`;
+      const columnName = `x_${ACCESSOR}${i}`;
```

Comment:
Accessors have to be named like this, at least for xy and heatmap charts. Otherwise, something breaks when navigating into Lens. 

It probably has something to do with some logic similar to what [`parse`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/public/datasources/form_based/operations/definitions/formula/parse.ts#L321) does. 

Perhaps because the accessor name matched with a regular column, ie:
accessor: metric_formula_column0_x
formula: metric_formula_column0

In such cases, the accessor was not treated as an accessor, but as a formula column. Starting with `x` or `y` we ensure this doesn't happen.


https://github.com/elastic/kibana/assets/2767137/400b2639-c6fa-4c32-8f7f-071e95b492ca